splunk/resource_splunk_authorization_roles_test.go

```diff
@@ -13,7 +13,8 @@ resource "splunk_authorization_roles" "role" {
   name = "new-role"
   default_app = "search"
   imported_roles = ["user"]
-  capabilities = ["accelerate_datamodel", "change_authentication"]
+  # capabilities are explicitly listed in a different order than returned by the Splunk API
+  capabilities = ["change_authentication", "accelerate_datamodel"]
```

Comment:
This change, without the change to using `TypeSet`, results in this failed acceptance test:

```
=== RUN   TestAccSplunkAuthorizationRoles
    testing.go:684: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: splunk_authorization_roles.role
          capabilities.#:                        "2" => "2"
          capabilities.0:                        "accelerate_datamodel" => "change_authentication"
          capabilities.1:                        "change_authentication" => "accelerate_datamodel"
          cumulative_realtime_search_jobs_quota: "0" => "0"
          cumulative_search_jobs_quota:          "0" => "0"
          default_app:                           "search" => "search"
          id:                                    "new-role" => "new-role"
          imported_roles.#:                      "1" => "1"
          imported_roles.0:                      "user" => "user"
          name:                                  "new-role" => "new-role"
          realtime_search_jobs_quota:            "6" => "6"
          search_disk_quota:                     "100" => "100"
          search_filter:                         "" => ""
          search_indexes_allowed.#:              "2" => "2"
          search_indexes_allowed.0:              "_audit" => "_audit"
          search_indexes_allowed.1:              "_internal" => "_internal"
          search_indexes_default.#:              "2" => "2"
          search_indexes_default.0:              "_audit" => "_audit"
          search_indexes_default.1:              "_internal" => "_internal"
          search_jobs_quota:                     "3" => "3"
          search_time_win:                       "-1" => "-1"
        
        
        
        STATE:
        
        splunk_authorization_roles.role:
          ID = new-role
          provider = provider.splunk
          capabilities.# = 2
          capabilities.0 = accelerate_datamodel
          capabilities.1 = change_authentication
          cumulative_realtime_search_jobs_quota = 0
          cumulative_search_jobs_quota = 0
          default_app = search
          imported_roles.# = 1
          imported_roles.0 = user
          name = new-role
          realtime_search_jobs_quota = 6
          search_disk_quota = 100
          search_filter = 
          search_indexes_allowed.# = 2
          search_indexes_allowed.0 = _audit
          search_indexes_allowed.1 = _internal
          search_indexes_default.# = 2
          search_indexes_default.0 = _audit
          search_indexes_default.1 = _internal
          search_jobs_quota = 3
          search_time_win = -1
--- FAIL: TestAccSplunkAuthorizationRoles (0.15s)
FAIL
FAIL    github.com/splunk/terraform-provider-splunk/splunk      0.629s
FAIL
```